### PR TITLE
Implement per-provider operator deployments in Helm

### LINKED ISSUE #11800: operator: Follow-up Helm changes for specialised operator images
#10972 stopped short of making changes to the Helm chart, but those will need to be made before 1.8 goes out of the door.

What needs to happen is that based on chose cloud-provider either of the images is selected:

- `quay.io/cilium/cilium-operator-aws`
- `quay.io/cilium/cilium-operator-azure`
- `quay.io/cilium/cilium-operator-generic`

I suppose if anyone wants classic `cilium/cilium-operator`, they'd be looking to override image directly, or perhaps it could be picked when one of the legacy `ipam=` modes is set, albeit I don't think we have those exposed in the chart at the moment.

### PR DESCRIPTION
**Pull request by joestringer**
Update the helm charts for for generic, aws and azure operator images.

Fixes: #11800

This is a simplification of the minimal required changes from #11837  for the v1.8.0 release.

### CODE REVIEW

**Comment by joestringer** [comment]
test-focus K8sUpdates

**Comment by joestringer** [comment]
test-focus K8sUpdates

**Review by aanm — APPROVED** [review]

**Comment by joestringer** [comment]
test-me-please

**Review by tgraf — APPROVED** [review]

**Review by tgraf — CHANGES_REQUESTED** [review]
The following needs to be changed as well:
```
docker tag "cilium/cilium:$2" "$1/cilium/cilium:$2"
docker tag "cilium/cilium:$2" "$1/cilium/cilium-dev:$2"
docker tag "cilium/operator:$2" "$1/cilium/operator:$2"
docker tag "cilium/hubble-relay:$2" "$1/cilium/hubble-relay:$2"
```

**Comment by tgraf** [comment]
test-focus K8sUpdates

**Comment by tgraf** [comment]
test-focus K8sUpdates

File: Makefile.docker [hunk 1]
```diff
@@ -12,9 +12,17 @@ docker-cilium-image-for-developers:
...
+	$(MAKE) docker-operator-image
+	$(MAKE) docker-operator-aws-image
+	$(MAKE) docker-operator-azure-image
+	$(MAKE) docker-operator-generic-image
+
+docker-image-unstripped: clean docker-image-no-clean-unstripped docker-plugin-image-unstripped docker-hubble-relay-image-unstripped
+	$(MAKE) docker-operator-image-unstripped
+	$(MAKE) docker-operator-aws-image-unstripped
+	$(MAKE) docker-operator-azure-image-unstripped
+	$(MAKE) docker-operator-generic-image-unstripped
```
Comment (joestringer) [inline Makefile.docker:25]: :) would've expected these to be largely equivalent. "If you didn't know, now you know"

:+1:

**Comment by joestringer** [comment]
Focus passed :tada:

**Comment by joestringer** [comment]
test-me-please

**Comment by tgraf** [comment]
This PR currently uses the brute-force method to resolve this. We should avoid building all operator images in the CI in a follow-up PR.

**Review by tgraf — APPROVED** [review]

**Comment by joestringer** [comment]
Hit known flake https://github.com/cilium/cilium/issues/10929

**Comment by joestringer** [comment]
The rest is green, merging.

**Pull request merged**